comps/router/src/integrations/controllers/semantic_router_controller/semantic_router_controller.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+
+# from decorators import log_latency
+from dotenv import load_dotenv
+from semantic_router import Route
+from semantic_router.encoders import HuggingFaceEncoder, OpenAIEncoder
+from semantic_router.routers import SemanticRouter
+
+from comps.cores.telemetry.opea_telemetry import opea_telemetry
+from comps.router.src.integrations.controllers.base_controller import BaseController
+
+load_dotenv()
+hf_token = os.getenv("HF_TOKEN", "")
+openai_api_key = os.getenv("OPENAI_API_KEY", "")
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+
+
+class SemanticRouterController(BaseController):
+    def __init__(self, config, api_key=None, model_map=None):
+        self.config = config
+        self.model_map = model_map or {}
+
+        # 1) grab provider + model mapping
```

Comment:
I'd remove `1)` since you have to keep track of them and your next one shows `4)`!